src/components/TransactionHistoryRows/TransactionDetails/ApprovalAmount.tsx

```diff
@@ -0,0 +1,63 @@
+import { MaxUint256 } from '@ethersproject/constants'
+import { AssetId } from '@shapeshiftoss/caip'
+import { TxMetadata } from '@shapeshiftoss/chain-adapters'
+import { useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { selectAssetById, selectMarketDataById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { Row } from './Row'
+import { Text } from './Text'
+
+export const ApprovalAmount = ({
+  assetId,
+  value,
+  isRevoke,
+  parser,
+}: {
+  assetId: AssetId
+  value: string
+  isRevoke: boolean
+  parser: TxMetadata['parser'] | undefined
+}) => {
+  const translate = useTranslate()
+
+  const approvedAsset = useAppSelector(state => selectAssetById(state, assetId))
+  const approvedAmount = bn(value).div(bn(10).pow(approvedAsset.precision)).toString()
+  const approvedAssetMarketData = useAppSelector(state => selectMarketDataById(state, assetId))
+
+  const amountOrDefault = useMemo(() => {
+    if (isRevoke) return translate(`transactionRow.parser.${parser}.revoke`)
```

Comment:
This is purely stylistic, but I recently came across the `switch (true)` pattern and loved id, and this seems like a really elegant place to use it as you'd get the fallthrough for the same return values.

E.g. from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch):

```ts
switch (true) {
  case isSquare(shape):
    console.log('This shape is a square.');
    // Fall-through, since a square is a rectangle as well!
  case isRectangle(shape):
    console.log('This shape is a rectangle.');
  case isQuadrilateral(shape):
    console.log('This shape is a quadrilateral.');
    break;
  case isCircle(shape):
    console.log('This shape is a circle.');
    break;
}
```

Reply:
Poor man's pattern matching in JS, love it!
Gave it a try locally and unfortunately we would lose the type narrowing of the current flow, which also turns out to be dependent on the exact order of the checks, so fall-through would not produce the parsing we want 😞

Reply:
Aw, it was worth a try - thanks for humoring me 😁